src/s1reader/s1_orbit.py

```diff
@@ -337,7 +399,267 @@ def get_orbit_file_from_list(zip_path: str, orbit_file_list: list) -> str:
             break
 
     if not orbit_file_final:
-        msg = 'No orbit file was found in the file list provided.'
+        msg = ('No single orbit file was found in the file list provided. '
+               'Attempting to find set of RESORB files that covers the time period.')
+        warnings.warn(msg)
+        orbit_file_final = get_resorb_pair_from_list(zip_path, orbit_file_list, concat_resorb)
+
+    if not orbit_file_final:
+        msg = ('No single orbit file was found in the file list provided.')
         warnings.warn(msg)
 
     return orbit_file_final
+
+
+def _covers_timeframe(orbit_file: str, t_start_stop_frame: list) -> bool:
+    '''
+    Check if `orbitfile` covers `t_start_stop_frame`
+    Copied from `get_orbit_file_from_list()` and modified
+
+    Parameter
+    ---------
+    orbit_file: str
+
+    t_start_stop_frame: list(datetime.datetime)
+
+    Returns
+    -------
+    _: Bool
+        `True` if the orbit file covers the time range; False otherwise
+    '''
+
+    # get file name and extract state vector start/end time strings
+    t_orbit_start, t_orbit_stop = os.path.basename(orbit_file).split('_')[-2:]
+
+    # strip 'V' at start of start time string
+    t_orbit_start = datetime.datetime.strptime(t_orbit_start[1:], FMT)
+
+    # string '.EOF' from end of end time string
+    t_orbit_stop = datetime.datetime.strptime(t_orbit_stop[:-4], FMT)
+
+    # check if:
+    # 1. swath start and stop time > orbit file start time
+    # 2. swath start and stop time < orbit file stop time
+    return all(t_orbit_start < t < t_orbit_stop for t in t_start_stop_frame)
+
+
+def get_resorb_pair_from_list(zip_path: str, orbit_file_list: list,
+                               concatenate_resorb: bool=False) -> list | str | None:
+    '''
+    Find if there are TWO RESORB files that covers [start - margin_start_time, end]
+    If found, try to concatenate
```

Comment:
```suggestion
    If found, try to concatenate them
```

Reply:
Docstring was revised accordingly. Thank you.